### PR TITLE
include Kotlin DSL samples for chapter base_plugin

### PR DESCRIPTION
**Pull request by jnizet**
Issue #6442

Signed-off-by: jnizet <jb@ninja-squad.com>

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/docs/src/samples/userguide/basePlugin/kotlin/settings.gradle.kts [hunk 1]
```diff
@@ -0,0 +1 @@
+rootProject.name = 'base-plugin-example'
```
Comment (mkobit) [inline subprojects/docs/src/samples/userguide/basePlugin/kotlin/settings.gradle.kts]: I think this should be double quotes `"`

Comment (jnizet) [reply subprojects/docs/src/samples/userguide/basePlugin/kotlin/settings.gradle.kts]: Of course. Good catch. I would have expected the integration tests to fail though, and they didn't. Did I miss something?

Comment (jnizet) [reply subprojects/docs/src/samples/userguide/basePlugin/kotlin/settings.gradle.kts]: Get it. I forgot to change the .conf file in order to execute both builds.

**Comment by eriwen** [comment]
Merged! Thank you @jnizet!

**Pull request merged**